test/model/internal_link_test.dart

```diff
@@ -451,4 +452,48 @@ void main() {
       });
     });
   });
+
+  group('narrowLink URL generation', () {
+    final streams = [
+      eg.stream(streamId: 1, name: 'check'),
+    ];
+
+    test('narrowLink should include a trailing slash in the path and before #', () async {
+      final store = await setupStore(realmUrl: realmUrl, streams: streams);
+      final message = eg.streamMessage();
+      final narrow = TopicNarrow.ofMessage(message);
+      final messageLink = narrowLink(store, narrow, nearMessageId: 12345);
+      final urlString = messageLink.toString();
+      final fragmentIndex = urlString.indexOf('#');
+      expect(fragmentIndex, greaterThan(0));
```

Comment:
We usually use [checks](https://pub.dev/packages/checks) for our tests. So this can be `check(fragmentIndex).isGreaterThan(0);` instead. This applies to the other checks here too.